apps/server/src/modules/user-login-migration/service/migration-check.service.ts

```diff
@@ -1,25 +1,34 @@
 import { Injectable } from '@nestjs/common';
-import { EntityId } from '@shared/domain';
+import { EntityId, UserLoginMigrationDO } from '@shared/domain';
 import { SchoolDO } from '@shared/domain/domainobject/school.do';
 import { UserDO } from '@shared/domain/domainobject/user.do';
+import { UserLoginMigrationRepo } from '@shared/repo/userloginmigration/user-login-migration.repo';
 import { SchoolService } from '@src/modules/school';
 import { UserService } from '@src/modules/user';
 
 @Injectable()
 export class MigrationCheckService {
-	constructor(private readonly userService: UserService, private readonly schoolService: SchoolService) {}
+	constructor(
+		private readonly userService: UserService,
+		private readonly schoolService: SchoolService,
+		private readonly userLoginMigrationRepo: UserLoginMigrationRepo
+	) {}
 
 	async shouldUserMigrate(externalUserId: string, systemId: EntityId, officialSchoolNumber: string): Promise<boolean> {
 		const school: SchoolDO | null = await this.schoolService.getSchoolBySchoolNumber(officialSchoolNumber);
 
-		if (school) {
+		if (school && school.id) {
+			const userLoginMigration: UserLoginMigrationDO | null = await this.userLoginMigrationRepo.findBySchoolId(
+				school.id
+			);
+
 			const user: UserDO | null = await this.userService.findByExternalId(externalUserId, systemId);
 
-			if (user && user.lastLoginSystemChange && school.oauthMigrationPossible) {
-				const hasMigrated: boolean = user.lastLoginSystemChange > school.oauthMigrationPossible;
+			if (user && userLoginMigration && user.lastLoginSystemChange && !userLoginMigration.closedAt) {
```

Comment:
```suggestion
			if (user?.lastLoginSystemChange && !userLoginMigration?.closedAt) {
```

Reply:
`!userLoginMigration?.closedAt` does not work since it means `!userLoginMigration && !userLoginMigration.closedAt`, so it would still have to be `userLoginMigration && !userLoginMigration.closedAt`.